src/UNL/VisitorChat/Util.php

```diff
@@ -0,0 +1,30 @@
+<?php
+namespace UNL\VisitorChat;
+
+class Util
+{
+    public static function isAllowedDomain($url, array $domains)
+    {
+        if (!$parts = parse_url($url)) {
+            //it isn't a valid URL, don't allow
+            return false;
+        }
+        
+        if (!isset($parts['host'])) {
+            //it isn't a valid host, don't allow
+            return false;
+        }
+
+        $regex = "";
+        foreach ($domains as $domain) {
+            $regex .= preg_quote($domain, ".-/") . "|";
```

Comment:
I think your syntax is off for the `preq_quote` function. The second param should be the single character that is used as the pattern delimiter in `preg_match`. You might have better luck with:

```php
array_walk($domains, function(&$domain) {
    $domain = preg_quote($domain, "/");
});
$regex = implode($domains, "|");
```